### DISCUSSION TITLE
How to change RX buffer size in HW Serial and USB Serial of Pico Arduino ide

### DISCUSSION THREAD

**Discussion by butaniravi** [earlephilhower/arduino-pico#119] [Q&A] [answered]
by quick check I came to conclude that RX buffer size of UART0(Serial1) is 32 bytes, unfortunately I need 64 bytes RX buffer for application on hand, unlike avr micros cant find clue on how to change it.

Please help..

Cheers 
Ravi

**Comment by earlephilhower** [comment]
Right now the HW serial units use the hardware RX FIFO.  They can't be expanded, obviously, but it would be possible to rewrite the `SerialUART` to use a software circular buffer of arbitrary length.  You;d need an IRQ handler on FIFO !empty to read from the HW port and to implement an IRQ/multicore safe buffer.

**Reply by Ing-Dom** [reply]
Recognized this fact (no ram buffer) also at the moment digging into the code.
Would be great if this would have been mentioned in the doc, do you accept PR for documentation ?

**Reply by earlephilhower** [reply]
Sure, throw one together!  Documentation is always the hard/last/least fun bit so any help is appreciated.

**Comment by earlephilhower** [comment, marked as answer]
Latest cores have this configuration
